### PR TITLE
Player book + minimap aspect ratio fix

### PR DESCRIPTION
**Pull request by bsxf-47**
Keep aspect ratio of the player book and minimap, plus configurable scale of the book and notes in interface options.

### CODE REVIEW

File: src/gui/Text.cpp [hunk 1]
```diff
@@ -305,6 +305,17 @@ static float smallTextScale(float scale) {
 	return scale > 1.0f ? scale * 0.7f + 0.3f : scale;
 }
 
+void ARX_Text_scaleBookFont(float scale) {
+	res::path file;
+	if(!getFontFile(file)) {
+		return;
+	}
+
+	Font * nFontInBook     = createFont(file, "system_font_book_size", 18, smallTextScale(scale));
```
Comment (dscharrer) [inline src/gui/Text.cpp]: The extra space before = is not needed here.

File: src/gui/Note.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ void Note::deallocate() {
 	// Don't bother actually deleting the textures, we'll just need them again!
 	background = NULL, prevPage = NULL, nextPage = NULL;
 	
-	allocatedForRatio = Vec2f_ZERO;
+	currentScale = 1.0f;
```
Comment (dscharrer) [inline src/gui/Note.cpp]: This should probably be 0 or it might not get allocated at all if bookScale == 1.

**Comment by dscharrer** [comment]
Alright, looks like there are multiple issues (tested with the updated PR)

1. Notes don't show up until you open the player book at least once (or start a new quest). This worked fine before this PR.
Before opening the book:

After opening the book:


2. [Missing quest book text](https://bugs.arx-libertatis.org/arx/issues/1019) seems to be an older regression in 1.2- dev

3. [Missing quest book background](https://bugs.arx-libertatis.org/arx/issues/1021) is an even older regression - reproducible in 1.1.2 at least

Other issues:

- Player book will be centered after starting a new game as opposed to loading a save → overlaps secondary inventory. It is enough to got to the new quest screen and then load a save to trigger this bug. This is a regression from this PR.

After resizing the window or changing the book scale:

Interestingly, the journal tab background and text is positioned correctly in this case, but the tab buttons and other tabs are centered:

After resizing the window or changing the book scale:


- Rects for dragging equipped armor and helmets. With this PR you cannot drag the left side of the left leg in the player book. This is a regression from this PR, although the rect was already tight before.

- Map marker icons ignore book scale.



- [Buttons on the new quest screen should be positioned relatively to the book.](https://bugs.arx-libertatis.org/arx/issues/1022)
. This is not technically a regression from this PR, but with the fixed book aspect ratio it would be nice to fix this as well.

File: src/gui/Note.h [hunk 3]
```diff
@@ -44,7 +44,7 @@ struct Note {
 	
 	Note()
 		: _type(Undefined)
-		, allocatedForRatio(Vec2f_ZERO)
+		, currentScale(1.0f)
```
Comment (dscharrer) [inline src/gui/Note.h]: Same here.

File: src/gui/MiniMap.cpp [hunk 4]
```diff
@@ -333,11 +333,11 @@ void MiniMap::showBookMiniMap(int showLevel) {
 		
 		if(showLevel == ARX_LEVELS_GetRealNum(m_currentLevel)) {
 			playerPos = computePlayerPos(zoom, showLevel);
-			start = Vec2f(490.f, 220.f) - playerPos;
+			start = Vec2f(rect.center()) - playerPos;
```
Comment (bsxf-47) [inline src/gui/MiniMap.cpp]: If I recall correctly, this was just a little bit off the center so I assumed that someone forgot to update the value or thought it looks better that way (hardly noticeable IMHO). The center of the rectangle is much more easily scalable.

File: src/gui/book/Book.cpp [hunk 5]
```diff
@@ -254,18 +254,16 @@ static void RenderBookPlayerCharacter() {
...
 	if (BOOKZOOM) {
 		
-		rec = Rect(s32((120.f + BOOKDEC.x) * g_sizeRatio.x), s32((69.f + BOOKDEC.y) * g_sizeRatio.y),
-				   s32((330.f + BOOKDEC.x) * g_sizeRatio.x), s32((300.f + BOOKDEC.y) * g_sizeRatio.y));
+		rec = Rect(Vec2i((Vec2f(120.f, 69.f) + BOOKDEC) * g_sizeRatio), Vec2i((Vec2f(330.f, 300.f) + BOOKDEC) * g_sizeRatio));
 		GRenderer->Clear(Renderer::DepthBuffer, Color::none, 1.f, 1, &rec);
 		
 		if(ARXmenu.currentmode != AMCM_OFF) {
-			Rect vp = Rect(Vec2i(s32(139.f * g_sizeRatio.x), 0), s32(139.f * g_sizeRatio.x), s32(310.f * g_sizeRatio.y));
+			Rect vp = Rect(Vec2i(Vec2f(139.f, 0.f) * g_sizeRatio), Vec2i((Vec2f(278.f, 310.f) * g_sizeRatio)));
```
Comment (bsxf-47) [inline src/gui/book/Book.cpp]: This was an oversight, thanks for spotting that.

File: src/gui/MiniMap.cpp (hunk 4 above)
Comment (dscharrer) [inline src/gui/MiniMap.cpp]: Ok, the previous value might have been a bug then. Just wanted to make sure it wasn't an oversight.

**Pull request closed without merging**